### PR TITLE
[netcore] Bump CoreFX version

### LINKED ISSUE #15154: [netcore] Make System.Text.Json.Serialization.Tests.ValueTests.ValueFail Pass
The test currently fails with:

```
System.Text.Json.Serialization.Tests.ValueTests.ValueFail [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.Text.Json.JsonException)
      Actual:   typeof(System.MemberAccessException): Cannot create an abstract class: System.Enum
      ---- System.MemberAccessException : Cannot create an abstract class: System.Enum
      Stack Trace:
           at System.Delegate.CreateDelegate(Type type, Object firstArgument, MethodInfo method, Boolean throwOnBindFailure, Boolean allowClosed)
           at System.Delegate.CreateDelegate(Type type, Object firstArgument, MethodInfo method, Boolean throwOnBindFailure)
           at System.Delegate.CreateDelegate(Type type, Object firstArgument, MethodInfo method)
           at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType)
           at System.Text.Json.Serialization.ReflectionEmitMaterializer.CreateConstructor(Type type)
           at System.Text.Json.Serialization.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
           at System.Text.Json.Serialization.JsonSerializerOptions.GetOrAddClass(Type classType)
           at System.Text.Json.Serialization.ReadStackFrame.Initialize(Type type, JsonSerializerOptions options)
           at System.Text.Json.Serialization.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
           at System.Text.Json.Serialization.JsonSerializer.ParseCore(String json, Type returnType, JsonSerializerOptions options)
           at System.Text.Json.Serialization.JsonSerializer.Parse[Enum](String json, JsonSerializerOptions options)
           at System.Text.Json.Serialization.Tests.ValueTests.<>c__DisplayClass9_0.<ValueFail>b__29()
        ----- Inner Stack Trace -----
           at System.Delegate.CreateDelegate(Type type, Object firstArgument, MethodInfo method, Boolean throwOnBindFailure, Boolean allowClosed)
           at System.Delegate.CreateDelegate(Type type, Object firstArgument, MethodInfo method, Boolean throwOnBindFailure)
           at System.Delegate.CreateDelegate(Type type, Object firstArgument, MethodInfo method)
           at System.Reflection.Emit.DynamicMethod.CreateDelegate(Type delegateType)
           at System.Text.Json.Serialization.ReflectionEmitMaterializer.CreateConstructor(Type type)
           at System.Text.Json.Serialization.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
           at System.Text.Json.Serialization.JsonSerializerOptions.GetOrAddClass(Type classType)
           at System.Text.Json.Serialization.ReadStackFrame.Initialize(Type type, JsonSerializerOptions options)
           at System.Text.Json.Serialization.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
           at System.Text.Json.Serialization.JsonSerializer.ParseCore(String json, Type returnType, JsonSerializerOptions options)
           at System.Text.Json.Serialization.JsonSerializer.Parse[Enum](String json, JsonSerializerOptions options)
           at System.Text.Json.Serialization.Tests.ValueTests.<>c__DisplayClass9_0.<ValueFail>b__29()
```

It appears that JsonSerializer.Parse<Enum>(...) generates invalid IL through DynamicMethod,
Mono throws "System.MemberAccessException: Cannot create an abstract class: System.Enum" from System.Reflection.Emit.DynamicMethod.CreateDelegate
, CoreCLR creates the delegate, but fails to execute it with "System.InvalidOperationException: Instances of abstract classes cannot be created.

The goal is to make the test pass.

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #15154

### CODE REVIEW

File: netcore/Makefile [hunk 1]
```diff
@@ -154,7 +154,7 @@ run-tests-corefx-%: prepare update-tests-corefx
 	cp corefx/restore/corefx-restore.runtimeconfig.dev.json corefx/tests/extracted/$*/xunit.console.runtimeconfig.dev.json
 	cp corefx/restore/corefx-restore.runtimeconfig.dev.json corefx/tests/extracted/$*/Microsoft.DotNet.RemoteExecutorHost.runtimeconfig.dev.json
 	cp corefx/restore/corefx-restore.dll corefx/tests/extracted/$*/corefx-restore.dll
-	sed -i -e 's/3.0.0\"/$(NETCOREAPP_VERSION)\"/g' corefx/tests/extracted/$*/*.runtimeconfig.json
+	sed -i -e 's/5.0.0\"/$(NETCOREAPP_VERSION)\"/g' corefx/tests/extracted/$*/*.runtimeconfig.json
```
Comment (akoeplinger) [inline netcore/Makefile]: we need to bump the assembly version in System.Private.CoreLib too.

File: netcore/CoreFX.issues.rsp [hunk 2]
```diff
@@ -676,18 +676,6 @@
...
-##  System.Text.Json.Tests
-####################################################################
-
-# The JSON value is not in a supported DateTime format
--noclass System.Text.Json.Serialization.Tests.ObjectTests
--noclass System.Text.Json.Serialization.Tests.ArrayTests
--nomethod System.Text.Json.Serialization.Tests.DictionaryTests.DictionaryOfClasses
--nomethod System.Text.Json.Serialization.Tests.CacheTests.PropertyCacheWithMinInputsFirst
--nomethod System.Text.Json.Serialization.Tests.CacheTests.PropertyCacheWithMinInputsLast
--nomethod System.Text.Json.Serialization.Tests.ValueTests.ValueFail
```
Comment (EgorBo) [inline netcore/CoreFX.issues.rsp]: The `requires precise GC` group of tests can also be removed 🙂 (at the end of the rsp file)

Comment (filipnavara) [reply netcore/CoreFX.issues.rsp]: I remove the ones you fixed. I'm still planning to do more pruning and cleanup after successful test run.

**Comment by filipnavara** [comment]
The netcore/ARM64 failures are flaky/broken tests that were failing even before the version bump.

**Comment by steveisok** [comment]
@filipnavara yeah, we're working on fixing that.

**Comment by filipnavara** [comment]
I saw that there is open PR for that. Just wanted to mention it to make you aware that I don't consider it a blocker for this PR.

**Comment by filipnavara** [comment]
@monojenkins build Linux AArch64 FullAOT+Interpreter
@monojenkins build Linux AArch64 FullAOT+Interpreter+LLVM
@monojenkins build Windows x64 FullAOT+LLVM

**Comment by filipnavara** [comment]
I forced a re-run to pick up the ARM64 CI fix. The following tests failed:
- Tests.Integration.DiscoveryTests.DiscoverAddinsWithCombinedCustomExportAndMetadataAttribute (flaky, already in the Windows exclusions)
- System.ComponentModel.Composition.MetadataTests.TestOptionalMetadataValueTypeMismatch (flaky)
- System.Diagnostics.TraceSourceTests.SwitchClassTests.PruneTest (likely new and ARM64 specific)

**Comment by filipnavara** [comment]
Ah, my bad... System.Diagnostics.TraceSourceTests.SwitchClassTests.PruneTest is missing "precise GC" condition in CoreFX. There's another PruneTest that already has it. I'll put it into exclusions.

File: netcore/CoreFX.issues.rsp (hunk 2 above)
Comment (filipnavara) [reply netcore/CoreFX.issues.rsp]: There are three tests left that are still missing the condition.

**Comment by akoeplinger** [comment]
The ARM64 failure was System.Transactions.Local.Tests which we've seen before, merging :)

**Pull request merged**